lib/SILOptimizer/Differentiation/PullbackCloner.cpp

```diff
@@ -2431,7 +2431,7 @@ bool PullbackCloner::Implementation::runForSemanticMemberGetter() {
 
   // Switch based on the base tangent struct's value category.
   // TODO(TF-1255): Simplify using unified adjoint value data structure.
-  switch (tangentVectorSILTy.getCategory()) {
+  switch (getTangentValueCategory(origSelf)) {
   case SILValueCategory::Object: {
     auto adjResult = getAdjointValue(origEntry, origResult);
```

Comment:
Note: hypothetically, there may be a similar "tangent value category mismatch" hole on this line where `getAdjointValue(origEntry, origResult)` is called before checking that `getTangentValueCategory(origResult) == SILValueCategory::Object`.

I can't think of a trigger for this in practice (semantic member getter pullback generation for `origSelf` with object tangent value category but `origResult` with address tangent value category), so I think it's fine to skip it for now.